recipes/sdl/3.x/conanfile.py

```diff
@@ -0,0 +1,253 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import is_apple_os
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, replace_in_file, rm, rmdir, copy
+from conan.tools.gnu import PkgConfigDeps
+from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout, CMakeDeps
+from conan.tools.env import Environment
+
+import os
+
+required_conan_version = ">=1.55.0"
+
+# Subsystems, CMakeLists.txt#L234
+_subsystems = [
+    ("audio", []),
+    ("video", []),
+    ("gpu", ["video"]),
+    ("render", ["video"]),
+    ("camera", ["video"]),
+    ("joystick", []),
+    ("haptic", ["joystick"]),
+    ("hidapi", []),
+    ("power", []),
+    ("sensor", []),
+    ("dialog", []),
+]
+
+class SDLConan(ConanFile):
+    name = "sdl"
+    description = "A cross-platform development library designed to provide low level access to audio, keyboard, mouse, joystick, and graphics hardware"
+    license = "Zlib"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.libsdl.org"
+    topics = ("sdl3", "audio", "keyboard", "graphics", "opengl")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+
+    options = {
+        **{
+            "shared": [True, False],
+            "fPIC": [True, False],
+        },
+        **{subsystem: [True, False] for subsystem, _ in _subsystems},
+        **{
+            "alsa": [True, False],
+            "pulseaudio": [True, False],
+            "sndio": [True, False],
+            "opengl": [True, False],
+            "opengles": [True, False],
+            "libudev": [True, False],
+        }
+    }
+
+    default_options = {
+        **{
+            "shared": False,
+            "fPIC": False,
+        },
+        **{subsystem: True for subsystem, _ in _subsystems},
+        **{
+            ## Audio
+            # Linux only
+            "alsa": False,
+            "pulseaudio": False,
+            "sndio": False,
+            ## Video
+            "opengl": True,  # TODO: Off by default in apple_os
+            "opengles": True,
+            ## Other
+            "libudev": True,
+        }
+    }
+
+    def config_options(self):
```

Comment:
I'd also throw a ConanException here if the os is Android because without Java classes, the compiled C library is useless.

SDL's CMake script compiles the java classes, so Android can be supported if you provide a jdk and a Android SDK (it needs android.jar).

Reply:
No problem. Documenting this, and raising an exception is always useful for the next person.

Reply:
We have been able to compile Android locally and run the test_package in an emulator, so my current idea is to let users build for Android if they have already set things up properly, but don't expect the recipe to fully provide it. Does that sound good to you?

Reply:
No, the test_package is a executable that runs in cli mode only. 
Android gui executables must be built as a shared library, and have the java classes.

I think you can add android support by addind a jdk and android-ndk to the (build) requirements.
If cmake finds those, it'll create a SDL3-3.x.y.jar archive.

Reply:
FYI, I added support for building the SDL tests (not the examples) as a single-arch-debug-apk to CMake (if it finds all dependencies, only tested on Linux, and is only for internal SDL use). This was done to avoid Gradle.

Reply:
> I think you can add android support by adding a jdk and android-ndk to the (build) requirements.

This is something that we expect users to add using their profiles instead of having recipes explicitly depend on those

Reply:
That's reasonable.
I expect the sdl3 conan recipe to fail in build or package when the java classes are not built.

Reply:
The issue is that this might be a bit hard to detect the case where the user has properly configured it on their side and when they have not

Reply:
You can detect by checking for a few `ANDROID_*` environment variables, and the presence of `javac` and `jar.
But that will break easily.
The pragmatic approach is to fail the build when the jar file is not created.
Don't overcomplicate things.